### PR TITLE
Configure docs.helm.sh to pull content from kubernetes/helm

### PR DESCRIPTION
**Pull request by flynnduism**
This modifies the `gulp` build process, to import and index the latest docs content from github.com/kubernetes/helm, and use it to generate the [docs.helm.sh](https://docs.helm.sh) website.

## Why?

* no need to insert TOML into [the source](github.com/kubernetes/helm), i.e. Helm devs don't have to change how they work on docs
* avoids breaking test on kubernetes/helm caused by changing the docs format
* avoids having Hugo elements in the docs at kubernetes/helm being overwritten whenever `make docs` is ran during development
* the build script will fetching the latest docs, no manual syncing

--  

## How?

* Updates gulp task with [gulp-git](https://www.npmjs.com/package/gulp-git)
* Replaces local docs content with the latest copy from [kubernetes/helm](https://github.com/kubernetes/helm)
* Indexes the site content locally, instead of requiring the [source](https://github.com/kubernetes/helm/tree/master/docs) to use a Hugo-friendly format
  _(i.e. stop depending on the inclusion of TOML metadata)_

  this is handled locally by:

  - (i) [using gulp](https://github.com/helm/helm-www/pull/47/commits/25270548340b3c01e12f0eb35376a7071419c87e#diff-7013c4e2ba6e10a18c92eec155f9c1d4R101) to create page categories, and
  - (ii) manually building the site navigation in the [config.yml](https://github.com/helm/helm-www/pull/47/commits/0efc09b22fc3a0dabbe36351f2b729e70a54294c#diff-9eeff327a6e03d2bf2ac9a555745cbf2L20)

### CODE REVIEW

**Comment by flynnduism** [comment]
Some issues to resolve before review:

- [ ] fix the title format in the topbar
- [ ] reorder the content in subsections
- [ ] using_helm/using_helm path not working
- [ ] find and replace (or redirect plugin) to update and repair internal links

**Comment by flynnduism** [comment]
Various tidyups, redirects, and bugs now addressed.
Ready for some 👀  - @vdice or @slack perhaps?

**Comment by vdice** [comment]
@flynnduism sure thing!  In addition to code review, deploying site off this branch somewhere locally and checking it out probably be the ticket?

**Comment by flynnduism** [comment]
@vdice yeah that would be awesome!

**Comment by flynnduism** [comment]
@vdice staging updated with working webfonts: [http://docs-helm-sh.staging.deis.com/](http://docs-helm-sh.staging.deis.com/)

To get them working here, Hugo's config.toml required the `baseurl` be set to `http://docs-helm-sh.staging.deis.com/` instead of the production url.

File: docs/config.toml [hunk 1]
```diff
@@ -2,74 +2,563 @@ baseurl = "https://docs.helm.sh/"
...
 publishdir = "app"
 
 [params]
     title = "Helm Docs"
     author = "Ronan Flynn-Curran"
     description = "Documentation for Helm - The Kubernetes Package Manager."
 
-canonifyURLs = "false"
-relativeURLs = "true"
+canonifyURLs = "true"
```
Comment (vdice) [inline docs/config.toml]: curious about the change to flip this and the `relativeURLs` values from previous?

File: docs/config.toml [hunk 2]
```diff
@@ -2,74 +2,563 @@ baseurl = "https://docs.helm.sh/"
...
+relativeURLs = "false"
 pluralizeListTitles = "false"
 plainIdAnchors = "true"
+defaultContentLanguage = "en"
+
 
 [[menu.main]]
+  name = "Home"
+  url = "https://docs.helm.sh"
```
Comment (vdice) [inline docs/config.toml]: perhaps related to comment above -- are we sure we want to hardcode the url here? (as opposed to it being relative/based on environment where deployed)

Comment (flynnduism) [inline docs/config.toml]: Opened this as a follow-up issue ([#49](https://github.com/helm/helm-www/issues/49)). 
Hugo makes it a bit tricky, but it should be doable.

**Pull request merged**